filebeat/input/filestream/internal/input-logfile/manager.go

```diff
@@ -164,11 +164,18 @@ func (cim *InputManager) Create(config *common.Config) (input.Input, error) {
 		ID             string        `config:"id"`
 		CleanTimeout   time.Duration `config:"clean_timeout"`
 		HarvesterLimit uint64        `config:"harvester_limit"`
-	}{ID: "", CleanTimeout: cim.DefaultCleanTimeout, HarvesterLimit: 0}
+	}{CleanTimeout: cim.DefaultCleanTimeout}
 	if err := config.Unpack(&settings); err != nil {
 		return nil, err
 	}
 
+	if settings.ID == "" {
+		cim.Logger.Warn("not setting an ID for the filestream input may lead to " +
+			"duplicated data. Filestream input without an ID will not be supported in a future release. " +
+			"Adding an ID to an existing input will read all files from the beginning",
```

Comment:
My suggestion for the wording here is something like :

```go
"Creating a filestream input without an ID, which may lead to duplicated data.
Filestream inputs will require an ID in a future release.
NOTE: Adding an ID to an existing input will cause all files to be re-read from the beginning."
```

Per my earlier comment, I dislike that the suggested user action to add an ID will definitely cause data to be re-read, while not using an ID "may" cause data to be re-read. 

It would be nice if we could come up with the user correction to prevent data duplication (or at least give them a way to filter it out easily).

Reply:
I agree. The release enforcing IDs will have a migration path without data duplication (or duplicating as least as possible), however until then it would be nice to warn users not to create new inputs without IDs. The tricky is to find a trade-off and how to communicate it.

7.17.1 FF is today, so maybe we only do the final solution.